lua/luasnip/loaders/from_vscode.lua

```diff
@@ -67,69 +76,76 @@ local function get_file_snippets(file)
 					dscr = parts.description or name,
 					wordTrig = ls_conf.wordTrig,
 					priority = ls_conf.priority,
+					snippetType = ls_conf.autotrigger and "autosnippet"
+						or "snippet",
 				}, body)
 
 				if session.config.loaders_store_source then
 					-- only know file, not line or line_end.
 					snip._source = source.from_location(file)
 				end
 
-				if ls_conf.autotrigger then
-					table.insert(auto_lang_snips, snip)
+				-- snippet can be provided to multiple filetypes via scope.
+				local ft_tables = {}
+
+				-- https://code.visualstudio.com/docs/editor/userdefinedsnippets#_language-snippet-scope
+				-- scope can have multiple components.
+				if parts.scope and not ignore_scope then
+					for _, scope in
+						ipairs(vim.split(parts.scope, ",", { plain = true }))
+					do
+						snippets_by_ft[scope] = snippets_by_ft[scope] or {}
+						table.insert(ft_tables, snippets_by_ft[scope])
+					end
 				else
-					table.insert(lang_snips, snip)
+					table.insert(ft_tables, snippets_by_ft[default_filetype])
+				end
+
+				for _, t in ipairs(ft_tables) do
+					table.insert(t, snip)
```

Comment:
Too puzzling. Can be rewritten to a much shorter form:

```lua
-- https://code.visualstudio.com/docs/editor/userdefinedsnippets#_language-snippet-scope
-- scope can have multiple components.
if parts.scope and not ignore_scope then
	for _, scope in
	ipairs(vim.split(parts.scope, ",", { plain = true }))
	do
		snippets_by_ft[scope] = snippets_by_ft[scope] or {}
		table.insert(snippets_by_ft[scope], snip)
	end
else
	table.insert(snippets_by_ft[default_filetype], snip)
end
```

Reply:
Ah, good point, yeah that's a bit more straightforward :+1: